src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -131,12 +132,20 @@
      */
     public static final String INDEX_CHECKSUM_ON_MERGE = "index.checksum_on_merge";
 
+    /**
+     * The maximum size the version map should grow to before issuing a refresh. Can be an absolute value or a percentage of
+     * the current index memory buffer (defaults to 25%)
+     */
+    public static final String INDEX_VERSION_MAP_SIZE = "index.version_map_size";
```

Comment:
I think we shouldn't. The only reason I added it is to be able to change it in the cases we suspect this is potentially the source of an issue, mostly to validate it's not.